File: src/main/java/tk/hiddenname/smarthome/service/task/impl/listener/Listener.java
package tk.hiddenname.smarthome.service.task.impl.listener;

import tk.hiddenname.smarthome.exception.UnsupportedTriggerObjectTypeException;
import tk.hiddenname.smarthome.model.task.trigger.objects.TriggerObject;

public interface Listener {

    void register(TriggerObject object) throws UnsupportedTriggerObjectTypeException;

    void update(TriggerObject object) throws UnsupportedTriggerObjectTypeException;

    void unregister();

    void trigger(boolean flag);
}

File: src/main/kotlin/tk/hiddenname/smarthome/service/task/impl/listener/Listener.kt
package tk.hiddenname.smarthome.service.task.impl.listener

import tk.hiddenname.smarthome.model.task.trigger.objects.TriggerObject
import javax.validation.constraints.NotNull

interface Listener {

    fun register(triggerObject: @NotNull TriggerObject)

    fun update(triggerObject: @NotNull TriggerObject)

    fun unregister()

    fun trigger(flag: @NotNull Boolean)
}